tespy/components/customs.py

```diff
@@ -0,0 +1,818 @@
+        if self.pr1.is_set:
+            self.vec_res[k] = (
+                    self.pr1.val * self.inl[0].p.val_SI -
+                    self.outl[0].p.val_SI)
+            k += 1
+
+        ######################################################################
+        # equations for specified pressure ratio at hot side 2
+        if self.pr2.is_set:
+            self.vec_res[k] = (
+                    self.pr2.val * self.inl[1].p.val_SI -
+                    self.outl[1].p.val_SI)
+            k += 1
+
+        ######################################################################
+        # equations for specified pressure ratio at cold side
+        if self.pr3.is_set:
+            self.vec_res[k] = (
+                    self.pr3.val * self.inl[2].p.val_SI -
+                    self.outl[2].p.val_SI)
+            k += 1
+
+        ######################################################################
+        # equations for specified zeta at hot side 1
+        if self.zeta1.is_set:
+            self.vec_res[k] = self.zeta_func()
+            k += 1
+
+        ######################################################################
+        # equations for specified zeta at hot side 2
+        if self.zeta2.is_set:
+            self.vec_res[k] = self.zeta2_func()
+            k += 1
+
+        ######################################################################
+        # equations for specified zeta at cold side
+        if self.zeta3.is_set:
+            self.vec_res[k] = self.zeta3_func()
+            k += 1
+
+        ######################################################################
+        # additional equations
+        self.additional_equations(k)
+
+    def additional_equations(self, k):
+        r"""
+        Calculates vector vec_res with results of additional equations for this
+        component.
+
+        Equations
+
+            **mandatory equations at outlet 1 of the hot side**
+
+            .. math::
+
+                0 = h_{1,out} - h\left(p, x=0 \right)\\
+                x: \text{vapour mass fraction}
+
+            **mandatory equations at outlet of the cold side**
+
+            .. math::
+
+                0 = h_{1,out} - h\left(p, x=1 \right)\\
+                x: \text{vapour mass fraction}
+
+        Returns
+        -------
+        vec_res : list
+            Vector of residual values.
+        """
+
+        ######################################################################
+        # equation for saturated liquid at hot side 1 outlet
+        if self.subcooling.val is False:
+            o1 = self.outl[0].to_flow()
+            self.vec_res[k] = o1[2] - h_mix_pQ(o1, 0)
+            k += 1
+
+        ######################################################################
+        # equation for saturated gas at cold side outlet
+        if self.overheating.val is False:
+            o3 = self.outl[2].to_flow()
+            self.vec_res[k] = o3[2] - h_mix_pQ(o3, 1)
+            k += 1
```

Comment:
See comment on lines 61-65, we can move this into the equations part.

Reply:
Moved to the equation part.